src/Microsoft.OData.Edm/ExtensionMethods/ExtensionMethods.cs

```diff
@@ -1366,6 +1366,31 @@ public static EdmTerm AddTerm(this EdmModel model, string namespaceName, string
             return term;
         }
 
+        /// <summary>
+        /// Resolves the annotations target and returns the target segments as Edm elements.
+        /// </summary>
+        /// <param name="model">The EdmModel.</param>
+        /// <param name="target">Target string containing segments separated by '/'. For example: "A.B/C/D.E/Func1(NS.T,NS.T2)/P1".</param>
+        /// <returns>The created enumeration of <see cref="IEdmElement"/>.</returns>
+        public static IEnumerable<IEdmElement> ResolveAnnotationsTarget(this IEdmModel model, string target)
+        {
+            EdmUtil.CheckArgumentNull(model, nameof(model));
+            EdmUtil.CheckArgumentNull(target, nameof(target));
+
+            string[] targetSegments = target.Split('/');
+            int targetSegmentsCount = targetSegments.Length;
+            List<IEdmElement> results = new List<IEdmElement>();
+
+            IEdmVocabularyAnnotatable annotatable = AnnotationsTargetHelper.HandleTargetSegments(model, results, targetSegments, targetSegmentsCount, 0);

```

Comment:
Maybe `Resolve` instead of `Handle`? Maybe `ResolvePathSegments` or `ResolveEdmPathSegments`? Maybe `pathSegments` instead of `results`?

Reply:
Or:

public static IEdmAnnotationsTarget GetAnnotationsTarget(this IEdmModel model, string targetPath)